### PR TITLE
Add risk_score as a Node property and as an available Node field

### PR DESCRIPTION
**Pull request by rachkor**
Nexpose includes a risk score for each Node. This PR improves our plugin to pull in this data as a node property and as a new field in the plugin manager for the Node template.

### CODE REVIEW

File: spec/nexpose_upload_spec.rb [hunk 1]
```diff
@@ -134,12 +130,13 @@
...
     it "populates solutions regardless they are wrapped in paragraphs or lists" do
       expect(@content_service).to receive(:create_issue) do |args|
-        expect(args[:text]).to include("#[Solution]#\nApache HTTPD >= 2.0 and < 2.0.65")
+        expect(args[:text]).to include("#[Solution]#\n\nApache HTTPD >= 2.0 and < 2.0.65")
         OpenStruct.new(args)
       end.once
 
       expect(@content_service).to receive(:create_issue) do |args|
-        expect(args[:text]).to include("#[Solution]#\nYou can remove inode information from the ETag header")
+        expect(args[:text]).to include("#[Solution]#\n")
```
Comment (dormi) [inline spec/nexpose_upload_spec.rb]: Looks like we are taking a lot of undesired white space here.
I am looking into it.

File: lib/nexpose/node.rb [hunk 2]
```diff
@@ -18,7 +18,7 @@ def initialize(xml_node)
     def supported_tags
       [
         # attributes
-        :address, :device_id, :hardware_address, :site_name, :status,
+        :address, :device_id, :hardware_address, :site_name, :status, :risk_score,
```
Comment (dormi) [inline lib/nexpose/node.rb]: alpha order

File: lib/nexpose/node.rb [hunk 3]
```diff
@@ -73,7 +73,8 @@ def method_missing(method, *args)
       translations_table = {
         :device_id => 'device-id',
         :hardware_address => 'hardware-address',
-        :site_name => 'site-name'
+        :site_name => 'site-name',
+        :risk_score => 'risk-score'
```
Comment (dormi) [inline lib/nexpose/node.rb]: alpha order + new hash format

File: lib/nexpose/vulnerability.rb [hunk 4]
```diff
@@ -124,7 +124,9 @@ def cleanup_html(source)
       result.gsub!(/<URLLink LinkTitle=\"(.*?)\" LinkURL=\"(.*?)\"\/>/i) { "\"#{$1.strip}\":#{$2.strip} " }
       result.gsub!(/<URLLink LinkURL=\"(.*?)\" LinkTitle=\"(.*?)\"\/>/i) { "\"#{$2.strip}\":#{$1.strip} " }
       result.gsub!(/<URLLink(.*)LinkURL=\"(.*?)\"(.*?)>(.*?)<\/URLLink>/m) {|m| "\"#{$4.strip}\":#{$2.strip} " }
-
+      result.gsub!(/&gt;/, '>')
```
Comment (dormi) [inline lib/nexpose/vulnerability.rb]: @rachkor I was thinking on not adding this and change the specs to check for `&gt;`/`&lt;`. I say so because it looks like in Dradis we are already displaying this ok.

Comment (rachkor) [reply lib/nexpose/vulnerability.rb]: I had to add this because the `&gt;/&lt;` characters were displaying that way in Dradis (and breaking the specs in the process!)

File: dradis-nexpose.gemspec [hunk 5]
```diff
@@ -28,7 +28,7 @@ Gem::Specification.new do |spec|
   spec.add_dependency 'dradis-plugins', '~> 3.6'
   spec.add_dependency 'nokogiri', '~> 1.3'
 
-  spec.add_development_dependency 'bundler', '~> 1.6'
+  spec.add_development_dependency 'bundler', '>= 1.6'
```
Comment (etdsoft) [inline dradis-nexpose.gemspec]: @dormi why is this needed?

File: templates/full_node.fields [hunk 6]
```diff
@@ -6,4 +6,5 @@ node.names
 node.tests
 node.site_name
 node.status
-node.software
\ No newline at end of file
+node.software
+node.risk_score
```
Comment (etdsoft) [inline templates/full_node.fields]: :alpha

File: spec/nexpose_upload_spec.rb [hunk 7]
```diff
@@ -86,25 +86,20 @@
...
+        hash_including label: "1.1.1.1", type: :host
+      ).twice
+
       expect(@content_service).to receive(:create_note) do |args|
-        expect(args[:text]).to include("#[Host]#\n1.1.1.1")
+        expect(args[:text]).to include("#[Title]#\n1.1.1.1")
         expect(args[:node].label).to eq("1.1.1.1")
       end.once
 
-      expect(@content_service).to receive(:create_node) do |args|
```
Comment (etdsoft) [inline spec/nexpose_upload_spec.rb]: Why are we removing this part?

File: lib/nexpose/vulnerability.rb (hunk 4 above)
Comment (etdsoft) [reply lib/nexpose/vulnerability.rb]: Can you add a spec for the HTML entities please?

File: dradis-nexpose.gemspec (hunk 5 above)
Comment (dormi) [reply dradis-nexpose.gemspec]: when running specs, if bundler 1.6 is not installed it complains. we don't need that constraint.

File: lib/nexpose/vulnerability.rb (hunk 4 above)
Comment (dormi) [reply lib/nexpose/vulnerability.rb]: https://github.com/dradis/dradis-nexpose/pull/19/commits/d6e1a93abe84a9bb8d6a2b2f62e1d367c9c4a8eb

File: spec/nexpose_upload_spec.rb (hunk 7 above)
Comment (dormi) [reply spec/nexpose_upload_spec.rb]: Looks like creating the Definitions node was removed before: https://github.com/dradis/dradis-nexpose/pull/11 (https://github.com/dradis/dradis-nexpose/pull/11/commits/fc99612b2853ab216dd66ad2d40b72ec95925e4e) but the spec was still here (failing)

**Comment by etdsoft** [comment]
@dormi what's the purpose of the last commit?

I tried uploading `full.xml` to the `master` branch, and it already creates the Issue with the right Solution field (escaped HTML entities).

**Comment by dormi** [comment]
@etdsoft I think it is the other way around: we intend to transform escaped html entities into the real thing.

In `master` when we have html entities `&gt;` and `&lt:`, we are not transforming them, so we import this:
Captura de pantalla 2019-07-22 a las 13 26 40

In the `risk-score` branch we transform these 2 html entities into `<` and `>`. So we import this:
Captura de pantalla 2019-07-22 a las 13 47 05

The reason for this change is 
https://github.com/dradis/dradis-nexpose/pull/19#discussion_r301642116
> I had to add this because the &gt;/&lt; characters were displaying that way in Dradis (and breaking the specs in the process!)

(and it's true, when editing the imported issue in `master`, if in the xml it had `&lt;`, when editing the issue we see `&lt;` (even it looks like a correct `<` when just viewing the issue).
